docs/docs/features/dongle.mdx

```diff
@@ -0,0 +1,206 @@
+
+Benefits:
+
+- The "central" dongle results in increased battery life for the keyboard/former "central", as it is now a peripheral.
+- It is easier to connect to the host device.
+
+Disadvantages:
+
+- An extra microcontroller is needed.
+- The keyboard becomes unusable without the dongle.
+
+
+
+## Editing a Keyboard Definition to Support a Dongle
+
+
+In the examples below, we will be modifying the [New Keyboard Shield](../development/new-shield.mdx) guide files and will be building on top of that.
+
+We will add the dongle in a way that it can be easily enabled, but is disabled by default.
+
+### Keyboard Shield
+
+
+Lets start by adding the dongle body to the shield configuration.
+
+```kconfig title="Kconfig.shield"
+config SHIELD_MY_KEYBOARD_DONGLE
+    def_bool $(shields_list_contains,my_keyboard_dongle)
+```
+
+### Keyboard Defconfig
+
+
+<Tabs
+defaultValue="unibody"
+values={[
+{label: 'Unibody Keyboard', value: 'unibody'},
+{label: 'Split Keyboard', value: 'split'},
+]}
+>
+<TabItem value="unibody">
+
+Next we will modify the following lines
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_KEYBOARD_DONGLE
+
+    config ZMK_KEYBOARD_NAME
+        default "My Board"
+
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    config ZMK_SPLIT
+        default y
+
+    # Prevent the dongle from going to sleep
+    config ZMK_SLEEP
+        default n
+
+    # Increase the transmit power of the dongle
+    config BT_CLTR_TX_PWR_PLUS_8
+        default y
+
+endif
+
+if SHIELD_MY_KEYBOARD
+    config ZMK_KEYBOARD_NAME
+        default "My Board"
+    
+endif
+```
+
+</TabItem>
+<TabItem value="split">
+
+Next we will modify the following lines
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_KEYBOARD_DONGLE
+
+    config ZMK_KEYBOARD_NAME
+        default "My Keyboard"
+
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    config ZMK_SPLIT_BLE_CENTRAL_PERIPHERALS
+        default 2
+
+    # Increase the transmit power of the dongle
+    config BT_CLTR_TX_PWR_PLUS_8
+        default y
+
+endif
+
+
+if SHIELD_MY_KEYBOARD_LEFT
+
+    # Setting this to y will make the left half the central
+    # We want it to be the central when dongle is not used
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    # Check if the left half is the central
+    if ZMK_SPLIT_ROLE_CENTRAL
+        config ZMK_KEYBOARD_NAME
+            default "My Keyboard"
+            
+    endif # ZMK_SPLIT_ROLE_CENTRAL
+
+endif # SHIELD_MY_KEYBOARD_LEFT
+
+if SHIELD_MY_KEYBOARD_DONGLE || SHIELD_MY_KEYBOARD_LEFT || SHIELD_MY_KEYBOARD_RIGHT
+
+    config ZMK_SPLIT
+        default y
+
+endif
+```
+
+</TabItem>
+</Tabs>
+
+### Dongle Overlay
+
+We will replace the `kscan` with `kscan-mock` in the dongle overlay.
+
+```dts title="my_keyboard_dongle.overlay"
+// import the default shield configuration
+#include "my_keyboard.dtsi"
+
+&kscan {
+    compatible = "zmk,kscan-mock";
+
+    columns = <0>;
+    rows = <0>;
+    events = <0>;
+}
+
+```
```

Comment:
Note that only split keyboards typically have .dtsi files defined. It would be good to add a comment on the matrix transform being necessary. Your introductory sentence could be more specific about the effects of importing the file.

Reply:
added a comment